### ISSUE TITLE
ERROR: Can not provide predictions for the chosen molecule, see the list of possible molecules

### ISSUE THREAD

**Issue by JPFinnigan** [openvax/mhctools#34]
Hey guys, Alex asked me to run the mouse tumors through topiary to generated MHC-II predictions, but I'm having some difficulty running topiary w/ these settings. I've tried a bunch of different input styles for I-A (e.g H2-IAb, H-2-IAB, H-2-IAb) all of which return similar errors. 

```
WARNING:root:ERROR: Can not provide predictions for the chosen molecule, see the list of possible molecules

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/bin/topiary", line 6, in <module>
exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/scripts/topiary", line 64, in <module>
main()
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/scripts/topiary", line 46, in main
epitopes = predict_epitopes_from_args(args)
 File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/predict_epitopes.py", line 103, in predict_epitopes_from_args
raise_on_variant_effect_error=not args.skip_variant_errors)
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/predict_epitopes.py", line 76, in predict_epitopes
raise_on_variant_effect_error=raise_on_variant_effect_error)
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/mutant_epitope_predictor.py", line 309, in epitopes_from_variants
transcript_expression_threshold=transcript_expression_threshold)
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/mutant_epitope_predictor.py", line 212, in epitopes_from_mutation_effects
for x in self.mhc_model.predict(peptide_dict):
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/mhctools/mhctools/netmhcii_pan.py", line 104, in predict
raise ValueError("No epitopes from netMHCIIpan")
ValueError: No epitopes from netMHCIIpan
```

Full trace: 

```
johnfinnigan$ python /Library/Frameworks/Python.framework/Versions/2.7/bin/topiary --vcf ~/Desktop/TEMP/Results/Tumor_B16_F1_0821/VCF/MuTect/Tumor_B16_F1_0821.mutect.targets.pass.vcf --mhc-predictor netmhciipan --mhc-alleles H-2-IAB --ic50-cutoff 500 --mhc-epitope-lengths 15 --output-csv ~/Desktop/Tumor_B16_F1_0821.mutect.targets.pass.vcf.topiary.netMHCIIpan.csv
Topiary commandline arguments:
Namespace(ic50_cutoff=500.0, json_variant_files=[], keep_wildtype_epitopes=False, maf=[], mhc_alleles='H-2-IAB', mhc_alleles_file=None, mhc_epitope_lengths=[15], mhc_predictor='netmhciipan', output_csv='/Users/johnfinnigan/Desktop/Tumor_B16_F1_0821.mutect.targets.pass.vcf.topiary.netMHCIIpan.csv', output_html=None, padding_around_mutation=None, percentile_cutoff=None, reference_name=None, rna_gene_fpkm_file=None, rna_min_gene_expression=0.0, rna_min_transcript_expression=0.0, rna_remap_novel_genes_onto_ensembl=False, rna_transcript_fpkm_file=None, self_filter_directory=None, skip_variant_errors=False, variant=[], vcf=['/Users/johnfinnigan/Desktop/TEMP/Results/Tumor_B16_F1_0821/VCF/MuTect/Tumor_B16_F1_0821.mutect.targets.pass.vcf'])
INFO:root:Building MHC binding prediction type for alleles ['H-2-IAb'] and epitope lengths [15]
INFO:root:Silent mutation filtering removed 6769/10282 entries of EffectCollection
INFO:root:netMHCIIpan -length 15 -f /var/folders/mk/v3ggtgds0c5_j1grfmcylxj00000gn/T/peptidetw7dRp -a IA_b
INFO:root:netMHCIIpan finished with return code 0
INFO:root:Cleaning up /var/folders/mk/v3ggtgds0c5_j1grfmcylxj00000gn/T/netMHCIIpan_outputRlbWZb
INFO:root:Cleaning up /var/folders/mk/v3ggtgds0c5_j1grfmcylxj00000gn/T/peptidetw7dRp
WARNING:root:ERROR: Can not provide predictions for the chosen molecule, see the list of possible molecules

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/bin/topiary", line 6, in <module>
exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/scripts/topiary", line 64, in <module>
main()
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/scripts/topiary", line 46, in main
epitopes = predict_epitopes_from_args(args)
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/predict_epitopes.py", line 103, in predict_epitopes_from_args
raise_on_variant_effect_error=not args.skip_variant_errors)
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/predict_epitopes.py", line 76, in predict_epitopes
raise_on_variant_effect_error=raise_on_variant_effect_error)
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/mutant_epitope_predictor.py", line 309, in epitopes_from_variants
transcript_expression_threshold=transcript_expression_threshold)
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/topiary/topiary/mutant_epitope_predictor.py", line 212, in epitopes_from_mutation_effects
for x in self.mhc_model.predict(peptide_dict):
  File "/Users/johnfinnigan/Desktop/Utilities/Topiary/mhctools/mhctools/netmhcii_pan.py", line 104, in predict
raise ValueError("No epitopes from netMHCIIpan")
ValueError: No epitopes from netMHCIIpan
```

Class I predictions seem to be working just fine. Any idea what could be causing this? Also, it would be helpful if 

```
WARNING:root:ERROR: Can not provide predictions for the chosen molecule, see the list of possible molecules 
```

produced documentation w/h helps the user find the list of possible alleles/molecules. 

Thanks!

**Comment by iskandr** [comment]
@tavinathanson 

1) Can we print the command to see the alleles for a predictor 
2) The format is off for mouse allele names, it's currently " -a IA_b" but should probably be "-a H-2-IAb"

**Comment by JPFinnigan** [comment]
"-a IA_b" but should probably be "-a H-2-IAb"

Ok. I agree re: the latter specification. Thank's for clarifying, Alex!

**Comment by tavinathanson** [comment]
@JPFinnigan Just to make sure we're on the same page, the issue is that my code normalizes whatever you enter into "IA_b", which `netMHCIIpan` doesn't accept. I have the fix; should be in soon!

**Comment by tavinathanson** [comment]
@JPFinnigan The fix is now merged into `mhctools` version 0.1.7. Does it fix the problem for you?

**Comment by JPFinnigan** [comment]
@tavinathanson Looks like this resolved the issue. Thanks!
